Xamarin.Forms.Platform.Android/Renderers/ButtonRenderer.cs

```diff
@@ -55,13 +56,13 @@ public override SizeRequest GetDesiredSize(int widthConstraint, int heightConstr
 
 		protected override void OnLayout(bool changed, int l, int t, int r, int b)
 		{
-			if (_imageHeight > -1)
+			if (_imageHeight > -1 && Element.Padding.Bottom == 0 && Element.Padding.Top == 0)
```

Comment:
So I don't think just opting out like this quite works.  I've had success with this

```C#
protected override void OnLayout(bool changed, int l, int t, int r, int b)
		{
			if (_imageHeight > -1)
			{
				// We've got an image (and no text); it's already centered horizontally,
				// we just need to adjust the padding so it centers vertically
				var diff = ((b - Context.ToPixels(Element.Padding.Bottom + Element.Padding.Top))  - t - _imageHeight) / 2;
				diff = Math.Max(diff, 0);
				UpdateContentEdge(new Thickness(0, diff, 0, -diff));

			}
			else
			{
				UpdateContentEdge();
			}

			base.OnLayout(changed, l, t, r, b);
		}
```

Here's some tester code to try. Using the code you have if I start adding top/bottom padding it causes it to jump to jump as you add padding.

```C#
StackLayout layout = null;
			layout = new StackLayout()
			{
				Children =
					{
						new Button()
						{
							Image = "coffee.png",
							BackgroundColor = Color.Green,
							Padding = new Thickness(),
							HeightRequest = 400,
						},
						new Button()
						{
							Text = "Bottom",
							Command = new Command(() =>
							{
								var button = layout.Children[0] as Button;
								button.Padding = new Thickness(0, button.Padding.Top, 0,button.Padding.Bottom + 10);
								if(!String.IsNullOrWhiteSpace(button.Text))
									button.Text = "Top: " + button.Padding.Top.ToString() + " Bottom: " + button.Padding.Bottom.ToString();
							})
						},
						new Button()
						{
							Text = "Top",
							Command = new Command(() =>
							{
								var button = layout.Children[0] as Button;
								button.Padding = new Thickness(0, button.Padding.Top + 10, 0, button.Padding.Bottom);
								if(!String.IsNullOrWhiteSpace(button.Text))
									button.Text = "Top: " + button.Padding.Top.ToString() + " Bottom: " + button.Padding.Bottom.ToString();
							})
						},
						new Button()
						{
							Text = "Reset",
							Command = new Command(() =>
							{
								var button = layout.Children[0] as Button;
								button.Padding = new Thickness(0, 0, 0,0);
							})
						},
						new Button()
						{
							Text = "Set Text",
							Command = new Command(() =>
							{
								var button = layout.Children[0] as Button;
								button.Text = "some text";
							})
						},
						new Button()
						{
							Text = "Reset Text",
							Command = new Command(() =>
							{
								var button = layout.Children[0] as Button;
								button.Text = String.Empty;
							})
						},
					}
			};

			return new ContentPage()
			{
				Content = layout
			};
```